chemprop/cli/train.py

```diff
@@ -645,6 +649,101 @@ def build_splits(args, format_kwargs, featurization_kwargs):
     return train_data, val_data, test_data
 
 
+def format_number(num):
+    if 0.01 < np.abs(num) < 100:
+        return f"{num:.2f}"
+    else:
+        return f"{num:.2e}"
+
+
+def summarize(args, dataset: _MolGraphDatasetMixin) -> tuple[list, list]:
+    columns = get_column_names(
+        args.data_path,
+        args.smiles_columns,
+        args.reaction_columns,
+        args.target_columns,
+        args.ignore_columns,
+        args.splits_column,
+        args.weight_column,
+        args.no_header_row,
+    )
+
+    input_cols = (args.smiles_columns or []) + (args.reaction_columns or [])
+    target_cols = columns[1:] if len(input_cols) == 0 else columns[len(input_cols) :]
+    if args.task_type in ["regression", "regression-mve", "regression-evidential"]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+        y_mean = np.nanmean(y, axis=0)
+        y_std = np.nanstd(y, axis=0)
+        y_median = np.nanmedian(y, axis=0)
+        mean_dev_abs = np.abs(y - y_mean)
+        frac_1_sigma = np.sum((mean_dev_abs < y_std), axis=0) / np.count_nonzero(~np.isnan(y))
+        frac_2_sigma = np.sum((mean_dev_abs < 2 * y_std), axis=0) / np.count_nonzero(~np.isnan(y))
+
+        column_headers = ["Statistic"] + [f"Value ({target_cols[i]})" for i in range(y.shape[1])]
+        table_rows = [
+            ["Num. smiles"] + [f"{len(y)}" for i in range(y.shape[1])],
+            ["Num. targets"] + [f"{np.count_nonzero(~np.isnan(y))}" for i in range(y.shape[1])],
+            ["Num. NAN"] + [f"{np.count_nonzero(np.isnan(y))}" for i in range(y.shape[1])],
```

Comment:
```suggestion
        num_targets = np.sum(~np.isnan(y), axis=0)
        frac_1_sigma = np.sum((mean_dev_abs < y_std), axis=0) / num_targets
        frac_2_sigma = np.sum((mean_dev_abs < 2 * y_std), axis=0) / num_targets

        column_headers = ["Statistic"] + [f"Value ({target_cols[i]})" for i in range(y.shape[1])]
        table_rows = [
            ["Num. smiles"] + [f"{len(y)}" for i in range(y.shape[1])],
            ["Num. targets"] + [f"{num_targets[i]}" for i in range(y.shape[1])],
            ["Num. NaN"] + [f"{len(y) - num_targets[i]}" for i in range(y.shape[1])],
```
I now see that np.count_nonzero collapses the array to a single value instead of per column so that needs to be changed. Also I think NaN is preferred over NAN.